src/components/ui/dropdown/dropdown.js

```diff
@@ -144,7 +144,8 @@ const Dropdown = (props) => {
         onClosed();
       }
     }
-  });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedObject, selectedObjRef]);
```

Comment:
if we don't have this to limit the amount of time this useEffect gets called, everything here will get called around 3 times every time you switch a pipeline. And as a result, it will push to the history 3 times.

Reply:
but it does affect the test from line https://github.com/kedro-org/kedro-viz/blob/8d330b075ec34865fc0bd0bfe8a2d37fd5d94330/src/components/app/app.test.js#L120

I'm not sure if we need this test as it basically checks if the "Please select..." renders when the component renders. Should we check this after we select a different pipeline from the dropdown?

Reply:
Like I said above, I think we can remove this.